source/fem-dem/ib_particles_dem.cc

```diff
@@ -77,8 +140,14 @@ IBParticlesDEM<dim>::calculate_pp_contact_force(
 {
   for (auto &particle_one : dem_particles)
     {
-      for (auto &particle_two : dem_particles)
+      for (auto particle_contact_candidates_id =
+             particles_contact_candidates[particle_one.id].begin();
+           particle_contact_candidates_id !=
+           particles_contact_candidates[particle_one.id].end();
+           ++particle_contact_candidates_id)
         {
+          auto  particle_contact_id = *particle_contact_candidates_id;
+          auto &particle_two        = dem_particles[particle_contact_id];
```

Comment:
Why is one a reference and the other one a pointer?
If you are not modifying the particle, use const auto & instead of auto &